app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -198,8 +199,8 @@
       // TODO: follow-on PR to decide which of these events to toast,
       // via config in constants.js
       var eventWatchCallback = function(eventData) {
-        eventsByNameData = eventData.by('metadata.name');
-        eventsMap = formatAndFilterEvents(eventsByNameData);
+        eventsByUID = eventData.by('metadata.uid');
+        eventsMap = formatAndFilterEvents(eventsByUID);
```

Comment:
It's better to leave this as `eventsByName` unless we need it mapped by UID because `data.by('metadata.name')` is a no-op. `data.by('metadata.uid')` forces the map to be re-keyed.

Being a bit nit picky here because I'm afraid this might get called frequently.

Reply:
I'd make it a local var to this function if it's not used outside the function.

Reply:
I can swap it back here (though I think I'm going to turn it right back to uid in my next PR adding the other kind of notifications, but will see if I can leave it `.name` :smile: )

Reply:
OK that's fine then as long as it's needed